librz/core/disasm.c

```diff
@@ -1131,6 +1142,85 @@ static void ds_build_op_str(RDisasmState *ds, bool print_color) {
 				core->parser->subrel_addr = killme;
 			}
 		}
+
+		// add for str issue
+		if (begin_mark && markinst_num > 1 && strstr(ds->opstr, "0x")) {
+			//calculate: .got beginning addr - next inst addr
+			ut64 got_beginaddr = 0;
+			ut64 rodata_beginaddr = 0;
+			ut64 rodata_size = 0;
+			ut64 next_instaddr = 0;
+
+			RzList *sections;
+			sections = rz_bin_get_sections(core->bin);
+			RzBinSection *s;
+			RzListIter *iter;
+			rz_list_foreach (sections, iter, s) {
+				if (strcmp(s->name, ".got") == 0) {
+					got_beginaddr = s->vaddr;
+				}
+				if (strcmp(s->name, ".rodata") == 0) {
+					rodata_beginaddr = s->vaddr;
+					rodata_size = s->size;
+				}
+			}
+
+			char *numstr = (char *)malloc(sizeof(char) * 11);
+			memset(numstr, 0, sizeof(char) * 11);
+			strncpy(numstr, markinst[1], 10);
+			sscanf(numstr, "%llx", &next_instaddr);
+			ut64 offset_got_next = (got_beginaddr > next_instaddr) ? (got_beginaddr - next_instaddr) : (next_instaddr - got_beginaddr);
+			char offset_got_next_str[11] = { 0 };
+			snprintf(offset_got_next_str, 11, "0x%llx", offset_got_next);
```

Comment:
use `snprintf(offset_got_next_str, sizeof(offset_got_next_str), "0x%" PFMT64x, offset_got_next);`

Reply:
Thanks for your advice. I have updated the code.